game/src/main/scala/io/aigar/game/EntityContainer.scala

```diff
@@ -0,0 +1,22 @@
+package io.aigar.game
+
+import com.github.jpbetz.subspace.Vector2
+
+
+trait EntityContainer {
+  def respawn(grid: Grid, players: List[Player], respawnRetryAttempts: Int): Option[Vector2] = {
+    var newPosition = grid.randomPosition
```

Comment:
I don't think so, I think the result of the last `newPosition` is needed to return at the end of the function instead of `None` (if we failed too many times).